### PR TITLE
Update XCC toolchain docs and toolchain configuration files

### LINKED ISSUE #13680: XCC install directions need updating in boards/xtensa/intel_s1000_crb/doc/index.rst
The instructions for install XCC need some minor updates to be complete:

1. Missing how to setup/install the license file.

2. File name referenced is not correct:
`X6H3SUE_2016_4_linux_redist.tgz` should be: `2018-05-15_5afafc055d3f4_X6H3SUE_2016_4_linux_redist.tgz`

3. The tar command possible needs to be run with 'sudo'

### PR DESCRIPTION
**Pull request by nashif**
Contrary to all other toolchains, the xcc toolchain had a hardocded 
installation path. Replace this with XTENSA_TOOLCHAIN_PATH so users can 
install in different locations.

Update instructions to match latest SDK and avoid installation in system 
folders using sudo.

Fixes #13680

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by galak — COMMENTED** [review]
Still seeing license issues after following these directions:

```
The output was:
2
License checkout failed: No such feature exists.
Feature:       XTENSA_XCC_TIE
License path:  /home/galak/sue-creek-SDK-license.dat:/home/galak/xtensa/XtDevTools/install/tools/RF-2016.4-linux/XtensaTools/Tools/lic/license.dat:
FLEXnet Licensing error:-5,357
For further information, refer to the FLEXnet Licensing documentation,
available at "www.macrovision.com".
```

File: boards/xtensa/intel_s1000_crb/doc/index.rst [hunk 1]
```diff
@@ -84,44 +87,58 @@ Run the installer using these commands:
...
+After the tool chain is successfully installed, the core build needs to be
+installed as follows
 
 .. code-block:: console
 
-   tar -xvzf X6H3SUE_2016_4_linux_redist.tgz --directory /opt/xtensa/XtDevTools/install/builds
-   cd /opt/xtensa/XtDevTools/install/builds/RF-2016.4-linux/X6H3SUE_2016_4
-   sudo ./install
+   tar -xvzf 2018-05-15_5afafc055d3f4_X6H3SUE_2016_4_linux_redist.tgz --directory <path to xtensa SDK>/XtDevTools/install/builds
+   cd <path to xtens SDK>/XtDevTools/install/builds/RF-2016.4-linux/X6H3SUE_2016_4
```
Comment (galak) [inline boards/xtensa/intel_s1000_crb/doc/index.rst]: nit: xtens -> xtensa

File: boards/xtensa/intel_s1000_crb/doc/index.rst [hunk 2]
```diff
@@ -84,44 +87,58 @@ Run the installer using these commands:
...
+   ./install
 
-"install" is the Xtensa Processor Configuration Installation Tool which is required
-to update the installation path. When it prompts to enter the path to the Xtensa Tools
-directory, enter /opt/xtensa/XtDevTools/install/tools/RF-2016.4-linux/XtensaTools. You
-should use the default registry /opt/xtensa/XtDevTools/install/tools/RF-2016.4-linux/XtensaTools/config.
+The :file:`install` script is the Xtensa Processor Configuration Installation
+Tool which is required to update the installation path. When it prompts to
+enter the path to the Xtensa Tools directory, enter <path to
```
Comment (galak) [inline boards/xtensa/intel_s1000_crb/doc/index.rst]: quote these path's so they show up as fixed font in rendered docs.

**Comment by nashif** [comment]
> Still seeing license issues after following these directions:

hmm, no clue, works for me on multiple systems.

**Comment by sathishkuttan** [comment]
> > Still seeing license issues after following these directions:
> 
> hmm, no clue, works for me on multiple systems.

@galak can you check whether the network interface naming is the older scheme (eth0, eth1, etc.) or the newer scheme (enp0s2 etc.)? The Flexlm license manager only works with eth0, eth1 naming...

@nashif we should add this to the doc.

File: boards/xtensa/intel_s1000_crb/doc/index.rst [hunk 3]
```diff
@@ -84,44 +87,58 @@ Run the installer using these commands:
...
-package which is preventing successful installation, most probably gtk2-i686.  You can
-install this missing package with:
+Please note a dialogue box should pop-up after running this command. In case the
+graphical installation tool does not start, the tool will revert to console
+based installation. The graphical tool is the preferred installation method.
 
-.. code-block:: console
+If the graphical tool does not start, it means your system is missing some
+packages which is preventing successful installation, most probably
+``gtk2-i686``.  You can install any missing packages with::
```
Comment (sathishkuttan) [inline boards/xtensa/intel_s1000_crb/doc/index.rst]: The Xtensa installer and tools are 32-bit binaries. It's worth adding instructions to install 32-bit compatibility libraries in a 64-bit Linux development host.

File: boards/xtensa/intel_s1000_crb/doc/index.rst [hunk 4]
```diff
@@ -68,12 +68,15 @@ have a registered account at https://tensilicatools.com.
...
+Select version RF-2016.4 and download the archive. The archive contains two files:
+
+- Installer: :file:`Xplorer-6.0.4-linux-installer.bin` and
+- Core configuration
+  :file:`2018-05-15_5afafc055d3f4_X6H3SUE_2016_4_linux_redist.tgz`
 
 For JTAG based debugging, download the XOCD package as well.
 
-A node locked license key can also be generated from the tensilicatools.com portal.
+A node locked license key can also be generated from the SDK portal.
```
Comment (sathishkuttan) [inline boards/xtensa/intel_s1000_crb/doc/index.rst]: Add a link for `SDK portal` to the reference at the bottom?

**Review by sathishkuttan — COMMENTED** [review]
Setting up XOCD (configuring flyswatter2 via the `topology.xml` etc) would be a good addition to the doc as well.

**Comment by galak** [comment]
> @galak can you check whether the network interface naming is the older scheme (eth0, eth1, etc.) or the newer scheme (enp0s2 etc.)? The Flexlm license manager only works with eth0, eth1 naming...
> 
> @nashif we should add this to the doc.

Yes, this was related to the issue.  We should add something about eth0, and if you don't have eth0 when you setup the dummy interface, that the mac address needs to match the license file.  Which is what was causing me issues.

**Comment by nashif** [comment]
> Yes, this was related to the issue. We should add something about eth0, and if you don't have eth0 when you setup the dummy interface, that the mac address needs to match the license file. Which is what was causing me issues.

I am not going to mention how to workaround the license checker in the docs, this is out of scope for the document and might get us in trouble.

**Review by sathishkuttan — COMMENTED** [review]
Looks good. Please s/ensillica/ensilica (single l).

**Review by dbkinder — APPROVED** [review]

**Pull request merged**